core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -93,9 +100,22 @@ public OverwriteFiles caseSensitive(boolean isCaseSensitive) {
   }
 
   @Override
-  public OverwriteFiles validateNoConflictingAppends(Expression newConflictDetectionFilter) {
+  public OverwriteFiles conflictDetectionFilter(Expression newConflictDetectionFilter) {
     Preconditions.checkArgument(newConflictDetectionFilter != null, "Conflict detection filter cannot be null");
     this.conflictDetectionFilter = newConflictDetectionFilter;
+    return this;
+  }
+
+  @Override
+  public OverwriteFiles validateNoConflictingData() {
+    this.validateNewDataFiles = true;
+    failMissingDeletePaths();
+    return this;
+  }
+
+  @Override
+  public OverwriteFiles validateNoConflictingDeletes() {
+    this.validateNewDeleteFiles = true;
     failMissingDeletePaths();
```

Comment:
The fact that both of these have `failMissingDeletePaths` makes me think something happened with the cherry-picks. Investigating further.

Reply:
Actually, this is the way both of these functions look in master - Is this intentional?:

https://github.com/apache/iceberg/blob/fad77e8b4ee811f6bcbd1c440708e0bc5e1929c2/core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java#L100-L111

Wondering if this is correct.

Reply:
Nevermind. This looks correct. Here's the actual function called:

https://github.com/apache/iceberg/blob/fad77e8b4ee811f6bcbd1c440708e0bc5e1929c2/core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java#L143-L146

Reply:
Yes, this is intentional. This was to preserve the old behavior.